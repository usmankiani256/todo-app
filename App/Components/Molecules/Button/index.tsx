import * as React from 'react'
import { Button as PaperButton } from 'react-native-paper'

export type ButtonProps = React.ComponentProps<typeof PaperButton>

const Button = (props: ButtonProps) => {
  const { children, onPress, icon } = props

  return (
    <PaperButton {...props} icon={icon} mode="contained" onPress={onPress}>
      {children}
    </PaperButton>
  )
}

export default Button
