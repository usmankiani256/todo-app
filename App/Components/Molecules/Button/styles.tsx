import { Dimensions, Platform, StyleSheet } from 'react-native'

const { width } = Dimensions.get('window')

const useStyles = () =>
  StyleSheet.create({
    button: {
      width: width - 36,
    },
    buttonLabel: {
      paddingTop: Platform.select({ ios: 2 }),
    },
  })

export default useStyles
