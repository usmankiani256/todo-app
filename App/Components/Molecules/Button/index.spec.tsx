import * as React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import Button from './index'

const children = 'Click me!'

describe('Button Component', () => {
  it('renders correctly', () => {
    const rendered: any = render(<Button>{children}</Button>).toJSON()

    expect(rendered).toMatchSnapshot()
  })

  it('renders props', () => {
    const props = {
      icon: 'camera',
      disabled: true,
    }

    const rendered: any = render(
      <Button {...props}>{children}</Button>,
    ).toJSON()

    expect(rendered).toMatchSnapshot()
  })

  it('is called once on click', () => {
    const mockFn = jest.fn()

    const { getByText }: any = render(
      <Button onPress={mockFn}>{children}</Button>,
    )

    fireEvent.press(getByText(children))

    expect(mockFn).toBeCalledTimes(1)
  })
})
