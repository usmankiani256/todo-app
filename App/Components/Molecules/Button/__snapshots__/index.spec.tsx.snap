// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#6200ee",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 2,
      "minWidth": 64,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.75,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "width": 714,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "paddingTop": 2,
              },
            ],
          ]
        }
      >
        Click me!
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Component renders props 1`] = `
<View
  collapsable={false}
  nativeID="animatedComponent"
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.12)",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "width": 714,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "marginLeft": 12,
            "marginRight": -4,
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.32)",
                "fontSize": 16,
              },
              Array [
                Object {
                  "lineHeight": 16,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ó°„€
        </Text>
      </View>
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.32)",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "paddingTop": 2,
              },
            ],
          ]
        }
      >
        Click me!
      </Text>
    </View>
  </View>
</View>
`;
